### PR TITLE
Handle potential missing Location

### PR DESCRIPTION
**Pull request by vincent-psarga**
## Summary

With the introduction of `JavaMethod` for messages.SourceReference, we may get missing `Location` for some field, leading to failure in JSON formatter.

For `JavaMethod` field, it will now output `org.cucumber.ClassName.methodName` instead of an empty string (or potentially a RuntimeError)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] The change has been ported to Java.
- [ ] The change has been ported to Ruby.
- [ ] The change has been ported to JavaScript.
- [x] The change has been ported to Go.
- [ ] The change has been ported to .NET.
- [x] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated the CHANGELOG accordingly.

### CODE REVIEW

**Comment by vincent-psarga** [comment]
@mpkorstanje I was not exactly sure how you would handle the `JavaStackTraceElement` message, so I went for something simple in the JSON output: `filename:line` (or simply `filename` if the `Location` field is not set).

Would that be OK for you this way ?

File: json-formatter/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ This document is formatted according to the principles of [Keep A CHANGELOG](htt
 
 ### Changed
 
+* Handle the `JavaMethod` and `JavaStackTraceElement` fields in `SourceReference`:
+  - outputs `<className>.<methodName>` for `JavaMethod`
```
Comment (mpkorstanje) [inline json-formatter/CHANGELOG.md:17]: Should be `<className>.<methodName>(<parameter1>,<parameter2>, ect)`

File: json-formatter/CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,10 @@ This document is formatted according to the principles of [Keep A CHANGELOG](htt
...
+  - outputs `<fileName>:<line?>` for `JavaStackTraceElement`
```
Comment (mpkorstanje) [inline json-formatter/CHANGELOG.md:18]: Should be `<className>.<methodName>(<fileName>:<line>)`

File: json-formatter/CHANGELOG.md (hunk 1 above)
Comment (aslakhellesoy) [reply json-formatter/CHANGELOG.md:17]: I've fixed this @mpkorstanje

File: json-formatter/CHANGELOG.md (hunk 2 above)
Comment (aslakhellesoy) [reply json-formatter/CHANGELOG.md:18]: The `JavaStackTraceElement` message (your addition) doesn't have parameter information, so I can't add this at the moment.

**Pull request merged**